crytic_compile/platform/hardhat.py

```diff
@@ -132,7 +132,10 @@ def hardhat_like_parsing(
                         )
 
                     source_unit = compilation_unit.create_source_unit(path)
-                    source_unit.ast = info["ast"]
+                    try:
+                        source_unit.ast = info["ast"]
+                    except:
+                        source_unit.ast = info["legacyAST"]
```

Comment:
```suggestion
                    source_unit.ast = info.get("legacyAST", "ast")
```

Reply:
Good suggestion!

Reply:
Should we swap the "legacyAST" and "ast" so it attempts to look at the modern ast first?

Reply:
Yeah, that makes sense